product_packaging_level/models/product_packaging.py

```diff
@@ -126,11 +127,27 @@ def _onchange_package_type(self):
         if package_type_level_id:
             self.packaging_level_id = package_type_level_id
 
-    @api.onchange("packaging_level_id", "package_type_id", "name", "name_policy")
-    def _onchange_name(self):
-        new_name = self.name
-        if self.name_policy == "by_package_level":
-            new_name = self.packaging_level_id.display_name
-        elif self.name_policy == "by_package_type":
-            new_name = self.package_type_id.name
-        self.name = new_name
+    @api.depends(
+        "packaging_level_id",
+        "package_type_id",
+        "name",
+        "name_policy",
+    )
+    def _compute_name(self):
+        for rec in self:
+            new_name = rec.name
+            if rec.name_policy == "by_package_level":
+                new_name = rec.packaging_level_id.display_name
+            elif rec.name_policy == "by_package_type":
+                new_name = rec.package_type_id.name
+            rec.name = new_name
+
```

Comment:
Why do you want to set a name that will not be the same into the database depending on user's lang in use at record create?

Reply:
I got your points, but since we have set `required=True` for the `name` field, we still need this compute method to store the value into the database. Do you have any suggestions ?

Reply:
@lmignon branch updated, could you please take a look ?

Reply:
I don't get it if this issue is solved... if it depends on the lang it should not be stored UNLESS you make it translatable too. No?